File: Laboratorio 1/Fibonacci/Fibonacci_long.cpp
#include <iostream>

using namespace std;

long auxiliar;
long primero = 0;
long segundo = 1;
long i = 2;


int main() {
  cout << "0" << " " << 0 << endl;
  cout << "1" << " " << 1 << endl;
 
  while(segundo > 0){
        auxiliar = segundo;
        segundo += primero;
        primero = auxiliar;
        cout << i << " " << segundo << endl;
        i += 1;
       
  }
  cout << "Para una variable de tipo Long se produce un overflow para n = " << i - 1 << endl;
}

File: Laboratorio 1/Fibonacci/Fibonacci_long.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package fibonacci;

/**
 *
 * @author Pedro Valderrama
 */
public class Fibonacci_long {
    public static void main(String[] args) {
        // TODO code application logic here
        long auxiliar;
        long primero = 0;
        long segundo = 1;
        long n = 2;
        System.out.printf("0 %s\n",primero);
        System.out.printf("1 %s\n",segundo);
   
        while(segundo > 0){
          auxiliar = segundo;
          segundo += primero;
          primero = auxiliar;
          System.out.printf("%s %s\n",n, segundo);
          n += 1;
    }
    System.out.printf("Para una variable de tipo Long se produce un overflow para n = %s\n", n-1);
    }
}
